### PR TITLE
Scripts/SpellGeneric: use modern C++ array instead of C array

### PR DESCRIPTION
**Pull request by FrancescoBorzi**
##### CHANGES PROPOSED:

-  Scripts/SpellGeneric: use modern C++ array instead of C array
-  this also fixes a warning from cppcheck (static analysis too):

> [src/server/scripts/Spells/spell_generic.cpp:83]: (error) Array '_itemId[3]' accessed at index 3, which is out of bounds.

##### TESTS PERFORMED:

- Compiles under Linux

##### HOW TO TEST THE CHANGES:

Just try to use a few spells to sure the generic system spell is not broken

##### Target branch(es):

Master

### CODE REVIEW

**Comment by FrancescoBorzi** [comment]
Currently blocked by: https://github.com/azerothcore/azerothcore-wotlk/issues/1259

File: src/server/scripts/Spells/spell_generic.cpp [hunk 1]
```diff
@@ -80,7 +79,7 @@ class spell_gen_model_visible : public SpellScriptLoader
         }
 
     private:
-        uint32 _itemId[3];
+        std::array<uint32, 3> _itemId = {0, 0, 0};
```
Comment (Viste) [inline src/server/scripts/Spells/spell_generic.cpp]: get an error here 
```/home/viste/dev/azerothcore-wotlk/src/server/scripts/Spells/spell_generic.cpp:82:49: error: array must be initialized with a brace-enclosed initializer
         std::array<uint32, 3> _itemId = {0, 0, 0}; ```

fix it by ``` std::array<uint32, 3> _itemId = { {0, 0, 0} }; ```  becouse its not assignments as i see

Comment (FrancescoBorzi) [reply src/server/scripts/Spells/spell_generic.cpp]: What version of clang do you have?

Comment (Viste) [reply src/server/scripts/Spells/spell_generic.cpp]: $> clang --version                                                                                                                                                                                   18.6m  Пт 11:08
clang version 3.8.0-2ubuntu4 (tags/RELEASE_380/final)
Target: x86_64-pc-linux-gnu
Thread model: posix
InstalledDir: /usr/bin

Comment (Viste) [reply src/server/scripts/Spells/spell_generic.cpp]: with 6.0 all ok

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL check my last comment. I made it work on both 3.8 and 6.0.0 clang versions.

File: src/server/scripts/Spells/spell_generic.cpp [hunk 2]
```diff
@@ -80,7 +79,7 @@ class spell_gen_model_visible : public SpellScriptLoader
         }
 
     private:
-        uint32 _itemId[3];
+        std::array<uint32, 3> _itemId = { {0, 0, 0} };
```
Comment (FrancescoBorzi) [inline src/server/scripts/Spells/spell_generic.cpp]: @BarbzYHOOL this will work on 3.8 too.

Comment (BarbzYHOOL) [reply src/server/scripts/Spells/spell_generic.cpp]: :pray:

**Comment by BarbzYHOOL** [comment]
"Just make sure the generic system spell is not broken"

What do you mean exactly? who has to test this? a developer?

**Comment by FrancescoBorzi** [comment]
@BarbzYHOOL sorry it was a bit ambiguous, I reworded in:

> Just try to use a few spells to sure the generic system spell is not broken

**Comment by FrancescoBorzi** [comment]
Merged in Staging since contains generic changes that are hard to test properly and might affect a wide range of things.

Staging will be merged into master after some testing with test players.

**Pull request merged**